app/models/manageiq/providers/vmware/infra_manager/scanning/job.rb

```diff
@@ -0,0 +1,309 @@
+class ManageIQ::Providers::Vmware::InfraManager::Scanning::Job < VmScan
+  # Make updates to default state machine to take into account snapshots and VimBroker
+  def load_transitions
+    super.tap do |transitions|
+      transitions.merge!(
+        :start_snapshot     => {'before_scan'               => 'snapshot_create',
+                                'waiting_for_broker'        => 'snapshot_create'},
+        :snapshot_complete  => {'snapshot_create'           => 'check_host_credentials',
+                                'snapshot_delete'           => 'synchronizing'},
```

Comment:
@chessbyte I think there is an issue with the state transitions here, snapshot_complete should go from `:snapshot_create` to `:check_host_credentials` but it is calling [`call_synchronize]`(https://github.com/ManageIQ/manageiq-providers-vmware/pull/491/files#diff-8070c235d0dc1ad4b868b36e3a9d2928R257)

Reply:
Fixed